dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java

```diff
@@ -841,30 +895,105 @@ private void disableMenu(final boolean disable) {
         ds3TransferLeft.setDisable(disable);
     }
 
-    /**
-     * check if bucket contains files or folders
-     *
-     * @param bucketName
-     * @return true if bucket is empty else return false
-     */
-    private boolean checkIfBucketEmpty(final String bucketName) {
+    public String getSearchedText() {
+        return ds3PanelSearch.getText();
+    }
 
+    //Method for calculating no. of files and capacity of selected tree item
+    public void calculateFiles(final TreeTableView<Ds3TreeTableValue> ds3TreeTableView) {
```

Comment:
This method has too much nesting.  It should be split into smaller methods, or re-written to reduce the complexity of the operation.